presto-main/src/main/java/com/facebook/presto/sql/planner/PartitioningScheme.java

```diff
@@ -74,8 +104,10 @@ public PartitioningScheme(
         hashColumn.ifPresent(column -> checkArgument(outputLayout.contains(column),
                 "Output layout (%s) don't include hash column (%s)", outputLayout, column));
 
-        checkArgument(!replicateNulls || columns.size() <= 1, "Must have at most one partitioning column when nullPartition is REPLICATE.");
-        this.replicateNulls = replicateNulls;
+        requireNonNull(replication, "replication is null");
+        checkArgument(replication.isCompatibleWithColumns(columns.size()), "Replication %s cannot be used with %s partitioning columns", replication, columns.size());
+        this.replication = replication;
```

Comment:
It's there, 2 lines earlier. It's not in the same line as assignment, because there is also another check in between.